src/ccl_power.c

```diff
@@ -768,3 +774,60 @@ smoothed with a tophat filter of comoving size 8 Mpc/h
 double ccl_sigma8(ccl_cosmology *cosmo, int *status) {
   return ccl_sigmaR(cosmo, 8/cosmo->params.h, 1., status);
 }
+
+// Integrand for kNL integral
+static double kNL_integrand(double k,void *params) {
+  KNL_pars *par=(KNL_pars *)params;
+
+  double pk=ccl_linear_matter_power(par->cosmo,k, 1.,par->status);
+
+  return pk;
+}
+
+/* --------- ROUTINE: ccl_kNL ---------
+INPUT: cosmology, scale factor
+TASK: compute kNL, the scale for the non-linear cut
+*/
+double ccl_kNL(ccl_cosmology *cosmo,double a,int *status) {
+  if (!cosmo->computed_linear_power) {
+    *status = CCL_ERROR_LINEAR_POWER_INIT;
+    ccl_cosmology_set_status_message(
+      cosmo,
+      "ccl_power.c: ccl_sigmaR(): linear power spctrum has not been computed!");
+    return NAN;
+  }
+  if (!cosmo->computed_growth){
+    *status = CCL_ERROR_GROWTH_INIT;
+    ccl_cosmology_set_status_message(
+      cosmo,
+      "ccl_power.c: ccl_sigmaR(): growth factor splines have not been precomputed!");
+    return NAN;
+  }
+
+  KNL_pars par;
+  par.status = status;
+
+  par.cosmo=cosmo;
+  gsl_integration_cquad_workspace *workspace =  NULL;
+  gsl_function F;
+  F.function=&kNL_integrand;
+  F.params=&par;
+  double k_NL;
```

Comment:
call this variable PL_integral (or something else clear) for readability, because it only becomes k_NL at the end.

Reply:
Good point, will change.